ixmp4/core/optimization/parameter.py

```diff
@@ -0,0 +1,123 @@
+from datetime import datetime
+from typing import Any, ClassVar, Iterable
+
+import pandas as pd
+
+from ixmp4.core.base import BaseFacade, BaseModelFacade
+from ixmp4.data.abstract import Docs as DocsModel
+from ixmp4.data.abstract import Parameter as ParameterModel
+from ixmp4.data.abstract import Run
+from ixmp4.data.abstract.optimization import Column
+
+
+class Parameter(BaseModelFacade):
```

Comment:
They are extremely similar and to some extent, we could have `Parameter` inherit from `Table`. I didn't do that for now because I thought if we then go and update `Table`, it might not be obvious that `Parameter` changes, too. Maybe the cleanest option is to set up a `BaseOptimizationItem` somewhere that contains the common ground between `Table`, `Parameter`, `Variable`, and `Equation`, and have all of them inherit from there.

Reply:
Not sure if there is any feature in Table that we would *not* want to be inherited... So better to keep the codebase simple for now and extend later once this becomes relevant.

Reply:
That's not as straightforward as I thought it would be. Naively trying to extract the maximum number of properties to a common base, I receive
```bash
E   sqlalchemy.exc.NoForeignKeysError: Can't determine the inherit condition between inherited table 'optimization_baseoptimizationitemmodel' and inheriting table 'optimization_table'; tables have no foreign key relationships established.  Please ensure the inheriting table has a foreign key relationship to the inherited table, or provide an 'on clause' using the 'inherit_condition' mapper argument.
```
So it looks like if I set up columns in a common base class, sqlalchemy interprets that as a separate table and inheriting tables need to be linked via ForeignKeys, which is not what we want.